conjure-go-contract/codecs/binary.go

```diff
@@ -44,8 +44,8 @@ func (codecBinary) Decode(r io.Reader, v interface{}) error {
 	if closer, ok := r.(io.ReadCloser); ok {
 		defer func() { _ = closer.Close() }()
 	}
-	if _, err := io.Copy(w, r); err != nil {
-		return werror.Convert(err)
+	if bytes, err := io.Copy(w, r); err != nil {
+		return werror.Wrap(err, "Failed to copy all bytes while decoding: "+err.Error(), werror.SafeParam("bytesCopied", bytes))
```

Comment:
```suggestion
		return werror.Wrap(err, "Failed to copy all bytes while decoding", werror.SafeParam("bytesCopied", bytes))
```
Using `Wrap` the cause will already be emitted when the error is logged, so no need to make this a dynamic string (same below)